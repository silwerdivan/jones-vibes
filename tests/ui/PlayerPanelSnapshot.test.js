import { JSDOM } from 'jsdom';
import fs from 'fs';
import path from 'path';

describe('Player Panel Snapshot Test', () => {
  let dom;

  beforeAll(() => {
    const html = fs.readFileSync(path.resolve(__dirname, '../../index.html'), 'utf8');
    dom = new JSDOM(html);
  });

  it('should match the snapshot of player 1 panel', () => {
    const player1Panel = dom.window.document.querySelector('#player1-status-panel');
    expect(player1Panel.innerHTML).toMatchSnapshot();
  });

  it('should match the snapshot of player 2 panel', () => {
    const player2Panel = dom.window.document.querySelector('#player2-status-panel');
    expect(player2Panel.innerHTML).toMatchSnapshot();
  });
});
