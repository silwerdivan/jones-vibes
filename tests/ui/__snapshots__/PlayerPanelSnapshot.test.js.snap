// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Player Panel Snapshot Test should match the snapshot of player 1 panel 1`] = `
"
                <h2>Player 1</h2>
                <div class="stats">
                    <div class="stat-item"><span class="stat-label">Cash:</span> <span id="p1-cash" class="stat-value currency">$0</span></div>
                    <div class="stat-item"><span class="stat-label">Savings:</span> <span id="p1-savings" class="stat-value">$0</span></div>
                    <div class="stat-item"><span class="stat-label">Loan:</span> <span id="p1-loan" class="stat-value">$0</span></div>
                    <div class="stat-item"><span class="stat-label">Happiness:</span> <span id="p1-happiness" class="stat-value">0</span></div>
                    <div class="stat-item"><span class="stat-label">Education:</span> <span id="p1-education" class="stat-value">None</span></div>
                    <div class="stat-item"><span class="stat-label">Career:</span> <span id="p1-career" class="stat-value">Unemployed</span></div>
                    <div class="stat-item"><span class="stat-label">Car:</span> <span id="p1-car" class="stat-value">No</span></div>
                </div>
                <div class="time-remaining">Time Remaining: <span id="p1-time">0</span> hours</div>
            "
`;

exports[`Player Panel Snapshot Test should match the snapshot of player 2 panel 1`] = `
"
                <h2>Player 2 (AI)</h2>
                <div class="stats">
                    <div class="stat-item"><span class="stat-label">Cash:</span> <span id="p2-cash" class="stat-value currency">$0</span></div>
                    <div class="stat-item"><span class="stat-label">Savings:</span> <span id="p2-savings" class="stat-value">$0</span></div>
                    <div class="stat-item"><span class="stat-label">Loan:</span> <span id="p2-loan" class="stat-value">$0</span></div>
                    <div class="stat-item"><span class="stat-label">Happiness:</span> <span id="p2-happiness" class="stat-value">0</span></div>
                    <div class="stat-item"><span class="stat-label">Education:</span> <span id="p2-education" class="stat-value">None</span></div>
                    <div class="stat-item"><span class="stat-label">Career:</span> <span id="p2-career" class="stat-value">Unemployed</span></div>
                    <div class="stat-item"><span class="stat-label">Car:</span> <span id="p2-car" class="stat-value">No</span></div>
                </div>
                <div class="time-remaining">Time Remaining: <span id="p2-time">0</span> hours</div>
            "
`;
